drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/model/converters/column/impl/GridWidgetColumnFactoryImpl.java

```diff
@@ -38,48 +38,49 @@
+                             }
+                         });
+        this.converters.addAll(converters);
     }
 
     @Override
-    public void initialise( final GuidedDecisionTable52 model,
-                            final AsyncPackageDataModelOracle oracle,
-                            final ColumnUtilities columnUtilities,
-                            final GuidedDecisionTableView.Presenter presenter ) {
-        for ( BaseColumnConverter converter : converters ) {
-            converter.initialise( model,
-                                  oracle,
-                                  columnUtilities,
-                                  presenter );
+    public void initialise(final GuidedDecisionTable52 model,
+                           final AsyncPackageDataModelOracle oracle,
+                           final ColumnUtilities columnUtilities,
+                           final GuidedDecisionTableView.Presenter presenter) {
+        for (BaseColumnConverter converter : converters) {
+            converter.initialise(model,
+                                 oracle,
+                                 columnUtilities,
+                                 presenter);
         }
     }
 
     @Override
-    public GridColumn<?> convertColumn( final BaseColumn column,
-                                        final GuidedDecisionTablePresenter.Access access,
-                                        final GuidedDecisionTableView gridWidget ) {
-        for ( BaseColumnConverter converter : converters ) {
-            if ( converter.handles( column ) ) {
-                final GridColumn<?> uiColumn = converter.convertColumn( column,
-                                                                        access,
-                                                                        gridWidget );
-                if ( uiColumn instanceof BaseUiColumn ) {
-                    ( (BaseUiColumn) uiColumn ).setColumnResizeListener( ( double width ) -> column.setWidth( ( (int) width ) ) );
+    public GridColumn<?> convertColumn(final BaseColumn column,
+                                       final GuidedDecisionTablePresenter.Access access,
+                                       final GuidedDecisionTableView gridWidget) {
+        for (BaseColumnConverter converter : converters) {
+            if (converter.handles(column)) {
+                final GridColumn<?> uiColumn = converter.convertColumn(column,
+                                                                       access,
+                                                                       gridWidget);
+                if (uiColumn instanceof BaseUiColumn) {
+                    ((BaseUiColumn) uiColumn).setColumnResizeListener((double width) -> column.setWidth(((int) width)));
                 }
+                //Copy back UI Column width into Model as BaseColumn does not have a width set by default
+                column.setWidth((int) uiColumn.getWidth());
```

Comment:
Copy UI column width back into model so "caption" width for new tables is correct.